src/api/transactionSubmission/simulate.ts

```diff
@@ -30,9 +30,9 @@ export class Simulate {
    */
   @ValidateFeePayerDataOnSimulation
   async simple(args: {
-    signerPublicKey: PublicKey;
+    signerPublicKey: PublicKey | null;
     transaction: AnyRawTransaction;
-    feePayerPublicKey?: PublicKey;
+    feePayerPublicKey?: PublicKey | null;
```

Comment:
I think we need `null` (or something equivalent) to cover the new use case: 
`feePayerPublicKey` can have:
(1) `undefined` if it is not a fee-payer transaction,
(2) `somePublicKey` if it is a fee-payer transaction, and the public/auth key will be checked on-chain, or
(3) `null` if it is a fee-payer transaction, and the public/auth key check will be skipped on-chain.
Among these, (3) is the new use case.

Reply:
I see. imo, it is a bit weird (in the SDK context and in the JS dev ecosystem) to differentiate between `null` and `undefined` and specifically ask the dev to write `feePayerPublicKey: null` as it means something critical.

Following this change - what would be the cases a dev still wants to NOT skip the public/auth key check? Are we (i.e Aptos) will be going with skipping the check from now on? wondering if we can combine the 1 and 3 use cases and just skip the check by default.

If we still want to keep the check, I would suggest maybe to introduce a new optional property for the dev to provide if it is a fee-payer transaction, and the public/auth key check will be skipped on-chain, something like
```
skipFeePayerPublicKeyCheck?:boolean
```

Reply:
> Following this change - what would be the cases a dev still wants to NOT skip the public/auth key check? Are we (i.e Aptos) will be going with skipping the check from now on? wondering if we can combine the 1 and 3 use cases and just skip the check by default.

Using a simulation, a dev may want to validate the public key that they have. Otherwise, they have to actually submit a transaction. (1) is different from (2) and (3) in terms of transaction types (fee-payer txn vs. simple txn).

Reply:
> If we still want to keep the check, I would suggest maybe to introduce a new optional property for the dev to provide if it is a fee-payer transaction, and the public/auth key check will be skipped on-chain, something like
> skipFeePayerPublicKeyCheck?:boolean

This sounds good for the fee-payer case!

However, it seems not clear how to apply in the secondary signer case: https://github.com/aptos-labs/aptos-ts-sdk/pull/467/files#r1686804774

Reply:
the secondary signer case makes sense to me - thanks for clarifying